### PR TITLE
8327994: Update code gen in CallGeneratorHelper

### PR DESCRIPTION
**Pull request by JornVernee**
Update the code gen code in CallGeneratorHelper to reflect the latest state of the libTest(Downcall/Upcall)(Stack).c and shared.h files.

- The previous code wanted users to pipe stdout into a file. But, since we have 5 files that need to be created, and the names of those files is important too, I've changed the script to write to those files directly instead.
- I moved the definition of `S_FFFF` to libVarArgs.c where it's used, as it's not actually generated by CallGeneratorHelper.
- GitHub doesn't render the changes to some of the files, but those files only contain either white space changes (some missing spaces after `,`), and copyright header updates.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8327994](https://bugs.openjdk.org/browse/JDK-8327994): Update code gen in CallGeneratorHelper (**Enhancement** - P5)

### Reviewers
 * [Maurizio Cimadamore](https://openjdk.org/census#mcimadamore) (@mcimadamore - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18269/head:pull/18269` \
`$ git checkout pull/18269`

Update a local copy of the PR: \
`$ git checkout pull/18269` \
`$ git pull https://git.openjdk.org/jdk.git pull/18269/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18269`

View PR using the GUI difftool: \
`$ git pr show -t 18269`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18269.diff">https://git.openjdk.org/jdk/pull/18269.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18269#issuecomment-1994262448)

### CODE REVIEW

File: test/jdk/java/foreign/CallGeneratorHelper.java [hunk 1]
```diff
@@ -188,136 +192,152 @@ static <Z extends Enum<Z>> String sigCode(List<Z> elems) {
         return elems.stream().map(p -> p.name().charAt(0) + "").collect(Collectors.joining());
     }
 
-    static void generateStructDecl(List<StructFieldType> fields) {
+    private  static void generateStructDecl(PrintStream out, List<StructFieldType> fields) {
```
Comment (turbanoff) [inline test/jdk/java/foreign/CallGeneratorHelper.java:195]:
```suggestion
    private static void generateStructDecl(PrintStream out, List<StructFieldType> fields) {
```

**Review by mcimadamore — APPROVED** [review]
Nice cleanup. No changes in libTestDowncallStack.c (not even minor ones) ?

**Comment by JornVernee** [comment]
> No changes in libTestDowncallStack.c (not even minor ones) ?

No, there was a 'missing' space between the prefix parameters and the actual parameters of the stack variants, and between the arguments passed when that callback was called in the upcall lib (So, white space only). libTestDowncall.c didn't have any changes.

**Pull request closed without merging**